### PR TITLE
Migrate Groovy to KTS

### PR DESCRIPTION
**Pull request by android-dev-lxl**
My first KTS migration, please let me know of any changes needed. 

- Upgraded Gradle version
- Migrated Grovy to KTS

### CODE REVIEW

**Comment by johnshea** [comment]
1. Repo is missing the LICENSE file.
2. Update gradle/wrapper/gradle-wrapper.properties to at least 8.1.1

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+dependencies {
+
+    implementation ("androidx.core:core-ktx:1.10.1")
+    implementation ("androidx.activity:activity-compose:1.7.2")
+    implementation (platform("androidx.compose:compose-bom:2022.10.00"))
+    implementation ("androidx.compose.ui:ui")
+    implementation ("androidx.compose.ui:ui-graphics")
+    implementation ("androidx.compose.ui:ui-tooling-preview")
+    implementation ("androidx.compose.material3:material3")
+    androidTestImplementation (platform("androidx.compose:compose-bom:2022.10.00"))
```
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/build.gradle.kts:74]: This is the dependencies in the new template: 
should we be adding the dependencies that we do not need.

   ```
 implementation 'androidx.core:core-ktx:1.8.0'
    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.3.1'
    implementation 'androidx.activity:activity-compose:1.5.1'
    implementation platform('androidx.compose:compose-bom:2022.10.00')
    implementation 'androidx.compose.ui:ui'
    implementation 'androidx.compose.ui:ui-graphics'
    implementation 'androidx.compose.ui:ui-tooling-preview'
    implementation 'androidx.compose.material3:material3'
    testImplementation 'junit:junit:4.13.2'
    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
    androidTestImplementation platform('androidx.compose:compose-bom:2022.10.00')
    androidTestImplementation 'androidx.compose.ui:ui-test-junit4'
    debugImplementation 'androidx.compose.ui:ui-tooling'
    debugImplementation 'androidx.compose.ui:ui-test-manifest'
```

File: build.gradle.kts [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id 'com.android.application' version '8.0.0' apply false
-    id 'com.android.library' version '8.0.0' apply false
-    id 'org.jetbrains.kotlin.android' version '1.7.20' apply false
+    id("com.android.application") version "8.0.1" apply false
+    id("com.android.library") version "8.0.1" apply false
```
Comment (johnshea) [inline build.gradle.kts:20]: Update version

File: app/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+}
+
+dependencies {
+    implementation(platform("androidx.compose:compose-bom:2023.05.01"))
+    implementation("androidx.compose.material3:material3")
+    implementation("androidx.compose.ui:ui")
+    implementation("androidx.compose.ui:ui-graphics")
+    implementation("androidx.compose.ui:ui-tooling-preview")
+
```
Comment (johnshea) [inline app/build.gradle.kts:75]: Remove blank line

File: app/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    implementation("androidx.activity:activity-compose:1.7.2")
```
Comment (johnshea) [inline app/build.gradle.kts:76]: Move this line to just below the BOM implementation line.

File: app/build.gradle.kts (hunk 3 above)
Comment (android-dev-lxl) [reply app/build.gradle.kts:75]: Why these are not compose libraries

File: app/build.gradle.kts (hunk 4 above)
Comment (android-dev-lxl) [reply app/build.gradle.kts:76]: This is not a compose dependency, why should we group this under BOM?

**Comment by android-dev-lxl** [comment]
Waiting on the dependency order to be finalized

**Comment by johnshea** [comment]
Dependency order has been finalized. Please update your code and re-request a review when ready.

**Review by johnshea — COMMENTED** [review]
Dependency order has been finalized.

Please update your code and re-request a review when ready.

**Review by johnshea — APPROVED** [review]

**Review by Raen111 — COMMENTED** [review]
@android-dev-lxl

**Review by zaidusmanali — COMMENTED** [review]
Change

**Pull request merged**